services/ip_manager/src/main/java/com/futurewei/alcor/ipmanager/service/Ipv4AddrService.java

```diff
@@ -0,0 +1,420 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+package com.futurewei.alcor.ipmanager.service;
+
+import com.futurewei.alcor.ipmanager.http.Ipv4AddrRequest;
+import com.futurewei.alcor.ipmanager.http.Ipv4AddrRangeRequest;
+import com.futurewei.alcor.ipmanager.entity.Ipv4AddrState;
+import com.futurewei.alcor.ipmanager.entity.Ipv4AddrAlloc;
+import com.futurewei.alcor.ipmanager.entity.Ipv4AddrRange;
+import com.futurewei.alcor.ipmanager.http.Ipv4AddrRequestBulk;
+import com.futurewei.alcor.ipmanager.http.status.*;
+import com.futurewei.alcor.ipmanager.repo.Ipv4AddrRangeRepo;
+import com.futurewei.alcor.ipmanager.repo.Ipv4AddrRepo;
+import com.futurewei.alcor.utils.Ipv4AddrUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+@Service
+public class Ipv4AddrService {
```

Comment:
For IPv4AddrService, I would suggest to split into two layers, service layer and service impl layer.

Service Layer: Show the interface
Service Impl Layer: Show the implementation of the interface. This design allows different implementations later on.

One example of the above design is SubnetManager:

https://github.com/kevin-zhonghao/alcor/tree/feature/route_manager/services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service